server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1113,7 +1113,8 @@ private Owner setupOwner(Principal principal, String ownerKey) {
             log.warn("User {} does not have access to create consumers in org {}",
                 principal.getPrincipalName(), owner.getKey());
 
-            throw new NotFoundException(i18n.tr("owner with key: {0} was not found.", owner.getKey()));
+            throw new ForbiddenException(i18n.tr("{0} is not authorized to register with " +
```

Comment:
We still need to check if we're in the right condition to send this more explicit message. So far we only know that the principal is not allow to create consumers in the org, but we don't know if the user is part of the org.

Rather than replace the old message entirely, we should branch on whether or not the principal's owner list includes the owner in question (indicating the user is allowed to know about the org). The UserPrincipal has three methods for fetching the owners: ```getOwners```, ```getOwnerKeys```, and ```getOwnerIds```. The latter two simply iterate over the values returned by ```getOwners``` (which itself iterates over its permissions to build the list and does not check for uniqueness), so it's probably best to just use it directly and check if it contains any owner with the same ID as the owner we've fetched above on line 1104.

Reply:
agreed. fixed it.